components/raftstore-v2/src/operation/command/write/mod.rs

```diff
@@ -133,7 +134,10 @@ impl<EK: KvEngine, ER: RaftEngine> Peer<EK, ER> {
                 self.applied_to_current_term()
             };
             let (data, chs) = encoder.encode();
-            let res = self.propose(ctx, data);
+            let res = self
+                .validate_command(&header, None, &mut ctx.raft_metrics)
+                .and_then(|_| self.propose(ctx, data));
```

Comment:
How about `validate_command` before `encode`? So that you can skip cloning header.

Reply:
Also please add some comments about the bug and the fix.

Reply:
> Also please add some comments about the bug and the fix.

sure